source/code/include/playfab/PlayFabSettings.h

```diff
@@ -15,39 +15,22 @@ namespace PlayFab
         static const std::string sdkVersion;
         static const std::string buildIdentifier;
         static const std::string versionString;
-        static const std::string verticalName;
 
-        static const std::map<std::string, std::string> requestGetParams;
-
-        static bool useDevelopmentEnvironment;
-        static std::string developmentEnvironmentURL;
         static std::string productionEnvironmentURL;
-        static std::string titleId; // You must set this value for PlayFabSdk to work properly (Found in the Game Manager for your title, at the PlayFab Website)
         static ErrorCallback globalErrorHandler;
 
-        // Control whether all callbacks are threaded or whether the user manually controlls callback timing from their main-thread
+        static std::shared_ptr<PlayFabApiSettings> staticSettings;
```

Comment:
>shared_ptr [](start = 20, length = 10)

Any reason why this one is not unique pointer ? Will this be modified in any way ?

Reply:
It's not unique, because it will definitely be simultaneously multi-referenced by simultaneous API calls.
Its contents will be modified by customers, but presumably the pointer/obj itself should never change.
Perhaps const static shared_ptr?  The PlayFabApiSettings inside must be mutable though.

---
In reply to: [343312976](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343312976) [](ancestors = 343312976)

Reply:
Yeah, const pointer to mutable object works, so that's been updated.

---
In reply to: [343317288](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343317288) [](ancestors = 343317288,343312976)